static/js/protein/chart.ts

```diff
@@ -432,6 +460,280 @@ export function drawProteinInteractionChart(
+  */
+
+  const MAX_INTERACTIONS = 10;
+
+  const NODE_STYLE = {
+    circles: {
+      fillColors: ["mistyrose", "peachpuff", "lightCoral", "lightsalmon"],
+      radius: 15,
+      stroke: {
+        color: "#fff",
+        opacity: 1,
+        width: 1.5,
+      },
+    },
+    labels: {
+      font: {
+        color: "#222",
+        name: "public sans",
+        size: "8px",
+      },
+    },
+  };
+
+  const LINK_STYLE = {
+    length: 100,
+    stroke: {
+      color: "#999",
+      linecap: "round",
+      opacity: 0.6,
+      scoreWidthMultiplier: 8,
+    },
+  };
+
+  /*
+  DATA PROCESSING
+  Format post-processing:
+
+    nodes = [
+      { id: 1, name: "MECOM", species: "Human", depth: 0 },
+      { id: 2, name: "CtBP1", species: "Human", depth: 1 },
+      { id: 3, name: "SUPT16H", species: "Human", depth: 1 }
+    ]
+
+    links = [
+      { source: 1, target: 2, score: 0.3 },
+      { source: 1, target: 3, score: 0.7 }
+    ]
+  */
+
+  function nodeFromID(protein_speciesID: string, depth: number): ProteinNode {
+    // protein_speciesID: id of form {protein}_{species}, e.g. P53_HUMAN
+    const lastIndex = protein_speciesID.lastIndexOf("_"); // danger: assumes species name does not contain _
+    return {
+      depth,
+      id: protein_speciesID,
+      name: protein_speciesID.slice(0, lastIndex),
+      species: protein_speciesID.slice(lastIndex + 1),
+    };
+  }
+
+  // P53_HUMAN is central protein in below examples.
+  // take interaction names of the form P53_HUMAN_ASPP2_HUMAN | ASPP2_HUMAN_P53_HUMAN and parse into ASPP2_HUMAN.
+  const centerNodeID = data[0].parent;
+  let nodeData = data.map(({ name, value }) => {
+    // value is confidenceScore
+    let neighbor = "";
+    if (name.includes(`_${centerNodeID}`)) {
+      neighbor = name.replace(`_${centerNodeID}`, ""); // replace only first instance to handle self-interactions (P53_HUMAN_P53_HUMAN)
+    } else if (name.includes(`${centerNodeID}_`)) {
+      neighbor = name.replace(`${centerNodeID}_`, ""); // same here
+    }
+    const nodeDatum = nodeFromID(neighbor, 1);
+    nodeDatum["value"] = value;
+    return nodeDatum;
+  });
+
+  // delete duplicates and self-interactions (will add support for self-interactions later on)
+  const seen = new Set();
+  nodeData = nodeData.filter((node) => {
+    const duplicate = seen.has(node.name);
+    seen.add(node.name);
+    return !duplicate && node.id !== centerNodeID;
+  });
+
+  nodeData.sort((n1, n2) => n2.value - n1.value); // descending order of interaction confidenceScore
+  nodeData = nodeData.slice(0, MAX_INTERACTIONS); // consider only top 10 interactions to avoid clutter
+
+  const centerDatum = nodeFromID(centerNodeID, 0);
+  nodeData.push(centerDatum);
+
+  const linkData: InteractionLink[] = nodeData.map((node) => {
+    return {
+      score: node.value,
+      source: centerNodeID,
+      target: node.id,
+    };
+  });
+
+  const height = 400 - MARGIN.top - MARGIN.bottom;
```

Comment:
400 and 700 should be constants

Reply:
oops yes! replaced with Suhana's new constants GRAPH_HEIGHT_M, GRAPH_WIDTH_M.